### PR TITLE
Allow configuration of cluster-openstack and default-apps-openstack catalogs

### PR DESCRIPTION
**Pull request by tfussell**
Towards https://github.com/giantswarm/roadmap/issues/586

Adds two new flags `--cluster-app-catalog` and `--defaulta-apps-app-catalog` to simplify testing of apps in test catalogs.

Not adding a changelog entry as this adds functionality to a previous unreleased PR https://github.com/giantswarm/kubectl-gs/pull/635

Sample output with `kubectl-gs template cluster --provider openstack --cluster-topology --organization giantswarm-2 --cluster-app-version 1.2.3-abc --name thom1 --cluster-app-catalog giantswarm-test --default-apps-app-version 0.0.0-def --default-apps-app-catalog giantswarm-test`:
```
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: thom1
  namespace: org-giantswarm-2
spec:
  catalog: giantswarm-test
  config:
    configMap:
      name: ""
      namespace: ""
    secret:
      name: ""
      namespace: ""
  kubeConfig:
    context:
      name: ""
    inCluster: true
    secret:
      name: ""
      namespace: ""
  name: cluster-openstack
  namespace: org-giantswarm-2
  version: 1.2.3-abc
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: thom1-default-apps
  namespace: org-giantswarm-2
spec:
  catalog: giantswarm-test
  config:
    configMap:
      name: ""
      namespace: ""
    secret:
      name: ""
      namespace: ""
  kubeConfig:
    context:
      name: ""
    inCluster: true
    secret:
      name: ""
      namespace: ""
  name: default-apps-openstack
  namespace: org-giantswarm-2
  version: 0.0.0-def
```

### CODE REVIEW

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -129,10 +135,12 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.OpenStack.NodeCIDR, flagOpenStackNodeCIDR, "", "CIDR used for the nodes.")
 
 	// OpenStack App only.
+	cmd.Flags().StringVar(&f.ClusterApp.ClusterAppCatalog, flagClusterAppCatalog, "giantswarm-catalog", "Cluster App version to be installed. (OpenStack App CR only).")
+	cmd.Flags().StringVar(&f.ClusterApp.ClusterAppVersion, flagClusterAppVersion, "0.2.2", "Cluster App version to be installed. (OpenStack App CR only).")
 	cmd.Flags().StringVar(&f.ClusterApp.ClusterUserConfigMap, flagClusterUserConfigMap, "", "Path to the user values configmap YAML file for Cluster App (OpenStack App CR only).")
-	cmd.Flags().StringVar(&f.ClusterApp.ClusterAppVersion, flagClusterAppVersion, "0.1.0", "Cluster App version to be installed. (OpenStack App CR only).")
-	cmd.Flags().StringVar(&f.ClusterApp.DefaultAppsUserConfigMap, flagDefaultAppsUserConfigMap, "", "Path to the user values configmap YAML file for Default Apps App (OpenStack App CR only).")
+	cmd.Flags().StringVar(&f.ClusterApp.DefaultAppsAppCatalog, flagDefaultAppsAppCatalog, "giantswarm-catalog", "Default Apps App version to be installed. (OpenStack App CR only).")
```
Comment (tfussell) [inline cmd/template/cluster/flag.go:141]: Changing default to `giantswarm` catalog in https://github.com/giantswarm/default-apps-openstack/pull/8

File: cmd/template/cluster/flag.go [hunk 2]
```diff
@@ -129,10 +135,12 @@ func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&f.OpenStack.NodeCIDR, flagOpenStackNodeCIDR, "", "CIDR used for the nodes.")
 
 	// OpenStack App only.
+	cmd.Flags().StringVar(&f.ClusterApp.ClusterAppCatalog, flagClusterAppCatalog, "giantswarm-catalog", "Cluster App version to be installed. (OpenStack App CR only).")
```
Comment (tfussell) [inline cmd/template/cluster/flag.go:138]: Changed default to `giantswarm` catalog in https://github.com/giantswarm/cluster-openstack/pull/24

**Review by kopiczko — APPROVED** [review]

File: cmd/template/cluster/provider/capo.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ func WriteOpenStackTemplateAppCR(ctx context.Context, config ClusterCRsConfig) e
 
 	clusterAppConfig := templateapp.Config{
 		AppName:   config.Name,
-		Catalog:   "control-plane-catalog",
+		Catalog:   config.ClusterAppCatalog,
```
Comment (MarcelMue) [inline cmd/template/cluster/provider/capo.go:84]: Do we really need to make it all configurable? I mean it should be pretty simple to change the manifest after this command is run. I would personally just set the catalog here to `giantswarm` and keep the number of flags limited from a UX pov

Comment (tfussell) [reply cmd/template/cluster/provider/capo.go:84]: I agree about not needing other catalogs, but we do need the ability to specify `giantswarm-test` to test branches of `cluster-openstack` and `default-apps-openstack` before they are released.

Comment (tfussell) [reply cmd/template/cluster/provider/capo.go:84]: On the UX side, we are hiding all OpenStack flags for now. Even when we un-hide them, we can leave these hidden as they are only really needed for GS internal use.

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**